data-prepper-plugins/opensearch/src/main/java/com/amazon/dataprepper/plugins/sink/opensearch/OpenSearchSink.java

```diff
@@ -68,10 +70,13 @@
 
   private final Timer bulkRequestTimer;
   private final Counter bulkRequestErrorsCounter;
+  private final AtomicLong bulkRequestSizeBytes;
+
   public OpenSearchSink(final PluginSetting pluginSetting) {
     super(pluginSetting);
     bulkRequestTimer = pluginMetrics.timer(BULKREQUEST_LATENCY);
     bulkRequestErrorsCounter = pluginMetrics.counter(BULKREQUEST_ERRORS);
+    bulkRequestSizeBytes = pluginMetrics.gauge(BULKREQUEST_SIZE_BYTES, new AtomicLong());
```

Comment:
Is there a reason you use a `guage` here. It appears that you are only incrementing. If that is the case you should use a Counter

Reply:
I would recommend using a [summary metric](https://github.com/prometheus/client_java#summary) here. This is the metric type we have used in the HttpSource to measure payload size of the requests.